inc/init.php

```diff
@@ -201,6 +201,12 @@
 $settings['bbname'] = strip_tags($settings['bbname']);
 $settings['orig_bblanguage'] = $settings['bblanguage'];
 
+// Adjust a relative upload path to an absolute one
+if(my_substr($settings['uploadspath'], 0, 1) != "/")
+{
+	$settings['uploadspath'] = MYBB_ROOT.$settings['uploadspath'];
+}
+
```

Comment:
@euantorano was right to question Windows compatibility, and @Eldenroot had mentioned Windows while we worked together on this PR, and it does need to be handled better. PHP on Windows allows for absolute paths that begin not with a forward slash but with either a backslash or a drive letter followed by colon followed by backslash. I suggest then that we replace the above code with this (tested on Windows):

```php
// Adjust a relative upload path to an absolute one
$iswin = strtoupper(substr(PHP_OS, 0, 3)) === 'WIN';
$char1 = my_substr($settings['uploadspath'], 0, 1);
if($char1 != '/' && !($iswin && ($char1 == '\\' || preg_match('(^[a-zA-Z]:\\\\)', $settings['uploadspath']))))
{
	$settings['uploadspath'] = MYBB_ROOT.$settings['uploadspath'];
}
```

Reply:
Note: `PHP_OS` reports the OS that PHP was __built for__, not the OS that PHP is __running on__. These are usually the same thing and is probably acceptable. other options include `php_uname('s')` or checking for one of the Windows-specific constants.

Otherwise, this looks good to me (though the regex is missing the delimiters). Only very minor point is the regex doesn't really need a capture group and could just be `preg_match('#^[a-zA-Z]:\\\\#')`

Reply:
Yep, I think `PHP_OS` is acceptable for detecting Windows. If you've built PHP for Windows, you're not going to be running it on Linux/MacOS.

> though the regex is missing the delimiters

Globally enclosing parentheses are valid delimiters.

> the regex doesn't really need a capture group

It's not a capture group. Again, those parens are the delimiters. I prefer to use parentheses as delimiters because they don't impose any limits on the characters that can be used within the regex.

Reply:
> Yep, I think `PHP_OS` is acceptable for detecting Windows. If you've built PHP for Windows, you're not going to be running it on Linux/MacOS.
> 
> > though the regex is missing the delimiters
> 
> Globally enclosing parentheses are valid delimiters.
> 
> > the regex doesn't really need a capture group
> 
> It's not a capture group. Again, those parens are the delimiters. I prefer to use parentheses as delimiters because they don't impose any limits on the characters that can be used within the regex.

You learn something new every day! 🤣